examples/libs/algorand/asset-tx/main.ts

```diff
@@ -0,0 +1,99 @@
+import { DfnsWallet } from '@dfns/lib-algorand'
+import { DfnsApiClient } from '@dfns/sdk'
+import { AsymmetricKeySigner } from '@dfns/sdk-keysigner'
+import { Algodv2, encodeObj, makeAssetTransferTxnWithSuggestedParams, modelsv2, waitForConfirmation } from 'algosdk'
+
+import * as dotenv from 'dotenv'
+
+dotenv.config()
+
+const algod = new Algodv2(process.env.ALGORAND_NODE_API_KEY!, process.env.ALGORAND_NODE_URL!)
+
+const initDfnsWallet = async (walletId: string) => {
+  const signer = new AsymmetricKeySigner({
+    privateKey: process.env.DFNS_PRIVATE_KEY!,
+    credId: process.env.DFNS_CRED_ID!,
+    appOrigin: process.env.DFNS_APP_ORIGIN!,
+  })
+
+  const dfnsClient = new DfnsApiClient({
+    appId: process.env.DFNS_APP_ID!,
+    authToken: process.env.DFNS_AUTH_TOKEN!,
+    baseUrl: process.env.DFNS_API_URL!,
+    signer,
+  })
+
+  return DfnsWallet.init({
+    walletId: walletId,
+    dfnsClient,
+  })
+}
+
+const TESTNET_USDC_ASSET_ID = 10458941
+
+async function main() {
+  const receiverWallet = await initDfnsWallet(process.env.ALGORAND_RECEIVER_WALLET_ID!)
+  console.log('algorand receiver address: %s', receiverWallet.address)
+
+  const senderWallet = await initDfnsWallet(process.env.ALGORAND_SENDER_WALLET_ID!)
+  console.log('algorand sender address: %s', senderWallet.address)
+
+  // check that our receiver wallet has opted in for the asset
+  // otherwise, we need to opt in. (The receiver wallet needs at least 0.1 ALGO) 
+  if (!await optInForAsset(receiverWallet)) {
+    console.log("receiver didn't optin for the asset... creating optin transction")
+    const txHash = await SendUsdcTransferTxn(receiverWallet, receiverWallet.address, 0)
+    console.log(`optin transaction broadcasted: ${txHash}`)
+    await waitForConfirmation(algod, txHash, 4)
+    console.log("optin transaction confirmed")
+  }
+
+  // Send USDC
+  console.log("sending 1 USDC")
+  let txHash = await SendUsdcTransferTxn(senderWallet, receiverWallet.address, 1000000)
+  console.log(`USDC transfer broadcasted: ${txHash}`)
+
+  console.log("opt-out for the asset")
+  const data = await algod.getAssetByID(Number(TESTNET_USDC_ASSET_ID)).do()
+  const asset = modelsv2.Asset.from_obj_for_encoding(data)
+
+  txHash = await SendUsdcTransferTxn(receiverWallet, asset.params.creator, 0, asset.params.creator)
```

Comment:
why the creator, why not close out to the sender instead?

Reply:
In general, when a user wishes to opt-out of an asset, they send their assets back to the creator of that asset.

However, in this case, since we're running a full cycle (opt-in, send, opt-out), it may make sense to close out at the sender (which will rarely happen in "real life").

I can send the funds back to the sender so that we don't lose funds when we test the whole process.

Reply:
yep, that's the idea 😄  thanks 👍